### PR TITLE
23 refactor folder and types

### LINKED ISSUE #23: refactor folder and types
## Description

Refactor and clean up types, etc.
- look into adding a `shared` file in the root repository to store all shared types (used by client and server)
- move `parseResponse.ts` into the `utils` folder (seems more fitting?)
- replace `any` types on the UI with the `ResponseData` interface
  image

## Helpful Documentation

Please list any documentation or resources that may be helpful.

- [ ] Documentation A
- [ ] Documentation B

### PR DESCRIPTION
**Pull request by viiiiiicccky**
## Description

**Fixes #23**

- look into adding a shared file in the root repository to store all shared types (used by client and server)
- move parseResponse.ts into the utils folder
- replace any types on the UI with the ResponseData interface

## Documentation Used

Please list any documentation or resources that were referenced while making these changes.

- [ ] Documentation A
- [ ] Documentation B

## Issues Faced and Resolutions

Please describe any issues you encountered during development and how you resolved them.

- Issue: Description of the issue
    - Resolution: Steps taken to resolve the issue

## How To Test?

- tested by running server and client

## Screenshots (if applicable):

Please add any screenshots that illustrate your changes.

### CODE REVIEW

File: client/pages/backend.tsx [hunk 1]
```diff
@@ -14,8 +15,7 @@ const Backend = () => {
       setImg(`data:image/jpeg;base64,${data}`);
     };
 
-    // TODO: replace with proper type
-    const onApiResonse = (data: any) => {
+    const onApiResonse = (data: ResponseData) => {
```
Comment (natapokie) [inline client/pages/backend.tsx:18]: if the api returns "no person found" for example, does the parsing function return just a string or an obj?
we might need to add `data; ResponseData | string` to handle both cases?

**Review by natapokie — CHANGES_REQUESTED** [review]
looks great! thanks for adding this vicky, i see all the parsed data on the frontend too,
image

i left one comment, would you be able to check and make sure we're handling the return type for when no person is found?

File: client/pages/backend.tsx (hunk 1 above)
Comment (viiiiiicccky) [reply client/pages/backend.tsx:18]: yeah i think you are right. do you mind helping me fix that? I'm trying to figure out some git issues might take sometime. tysm!

Comment (viiiiiicccky) [reply client/pages/backend.tsx]: thanks for catching, changed!

**Review by natapokie — APPROVED** [review]
looks great vicky!

**Review by vicliu2001 — APPROVED** [review]
looks good

**Review by eriysd — APPROVED** [review]
lgtm

**Pull request merged**